presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -75,14 +75,28 @@ public Analysis analyze(Statement statement)
     }
 
     public Analysis analyze(Statement statement, boolean isDescribe)
+    {
+        Analysis analysis = analyzeSemnatic(statement, isDescribe);
+        aclCheck(analysis);
+        return analysis;
+    }
+
+    public Analysis analyzeSemnatic(Statement statement, boolean isDescribe)
     {
         Statement rewrittenStatement = StatementRewrite.rewrite(session, metadata, sqlParser, queryExplainer, statement, parameters, accessControl, warningCollector);
         Analysis analysis = new Analysis(rewrittenStatement, parameters, isDescribe);
         StatementAnalyzer analyzer = new StatementAnalyzer(analysis, metadata, sqlParser, accessControl, session, warningCollector);
         analyzer.analyze(rewrittenStatement, Optional.empty());
         analyzeForUtilizedColumns(analysis, analysis.getStatement());
+        return analysis;
+    }
 
-        // check column access permissions for each table
+    /**
+     * check column access permissions for each table
+     * @param analysis the Analysis that needs to check ACL for
+     */
+    public void aclCheck(Analysis analysis)
```

Comment:
Call it `checkAccessControl`. We don't usually use acronym or abbreviation in our codebase.